lib/oauther.ex

```diff
@@ -1,28 +1,46 @@
 defmodule OAuther do
   defmodule Credentials do
+    @type t :: %__MODULE__{
+      consumer_key:    String.t,
+      consumer_secret: String.t,
+      token:           String.t,
+      token_secret:    String.t,
+      method:          :hmac_sha1 | :rsa_sha1 | :plaintext
+    }
+
     defstruct [:consumer_key, :consumer_secret,
                :token, :token_secret, method: :hmac_sha1]
   end
 
+  @type pair_of_strings :: {String.t, String.t}
```

Comment:
Any reason why we're calling this `pair_of_strings` instead of something like `string_header`?

And actually, we could have the type `header/1` defined like

```elixir
@type header(value) :: {String.t, value}
```

and then define

```elixir
@type string_header :: header(String.t)
```

or just use `header(String.t)` directly.